task/control-plane-agent/src/mgs_sidecar.rs

```diff
@@ -664,6 +665,37 @@ impl SpHandler for MgsHandler {
     ) -> Result<&'static [u8], SpError> {
         self.common.get_caboose_value(key)
     }
+
+    fn reset_component_prepare(
```

Comment:
This appears to be the third in a series of three copy-pasted routines. It would be really nice not to copy-paste like this; what's the purpose of it?